manager/dispatcher/assignments.go

```diff
@@ -35,7 +36,9 @@ func newAssignmentSet(log *logrus.Entry, dp *drivers.DriverProvider) *assignment
 }
 
 func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID, t *api.Task) {
-	a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	if _, exists := a.tasksUsingDependency[mapKey]; !exists {
+		a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	}
```

Comment:
can you elaborate on why this change is needed?

Reply:
This is in the case there are multiple tasks for the same secret, in which case subsequent calls to `assignSecret` would overwrite the map in `a.tasksUsingDependency[mapKey]`.

Reply:
I would suggest adding this comment to the code @sirlatrom